cmd/zfs/zfs_iter.c

```diff
@@ -212,11 +213,58 @@ zfs_free_sort_columns(zfs_sort_column_t *sc)
 	}
 }
 
-int
-zfs_sort_only_by_name(const zfs_sort_column_t *sc)
+boolean_t
+zfs_sort_only_by_fast(const zfs_sort_column_t *sc)
 {
-	return (sc != NULL && sc->sc_next == NULL &&
-	    sc->sc_prop == ZFS_PROP_NAME);
+	if (sc == NULL) {
+		/* No sort means we don't need any extra properties */
+		return (B_TRUE);
+	}
```

Comment:
So, the comment is useful here, but the if-statement is not necessary, as we will just fall through the while-loop below and return B_TRUE anyway. Perhaps this comment could be part of a general comment above the function? Something like:
```
/*
 * Return true if all of the properties to be sorted are populated by
 * dsl_dataset_fast_stat(). Note that sc == NULL (no sort) means we
 * don't need any extra properties, so returns true.
 * /